### DISCUSSION TITLE
Q-table, columns from server response, how to attach header-click event?

### DISCUSSION THREAD

**Discussion by zvolsky** [quasarframework/quasar#11308] [General - Components / Directives / etc] [answered]
Hi friends,
I have universal list component based on Q-table. In created() I ask first time for table rows and the response has always different columns (based on the parameters sent from frontend).
So I create `columns` based on this first response.
Everything works.
Now I need to have a `click` event on each header.
How can I do it properly?

At this time I am able to solve it, but I don't think it is correct.
I set
`columns = [{name: 'somecolumn', headerClass: 'listHdr'}, ...]`
ie. listHdr class for all headers; and well: it exists properly.
Now the update() event run more times.
I am able to determine the run, where I have not the events initialized but I have (first time) the `.listHdr` elements accessible.
And just at this run I attach the events, using plain javascript `.addEventListener('click',..)`

How to do it better?
I cannot find some quasar event for this case.

**Comment by MilosPaunovic** [comment]
Could you go by with using header slot?
[https://quasar.dev/vue-components/table#header-slots](https://quasar.dev/vue-components/table#header-slots)

**Reply by zvolsky** [reply]
With header slot I have at least such issue that the first column with selection checkboxes no longer has its header ie. the checkbox for 'select/deselect all'. Or I don't know how to define it or keep it in header slot.

**Comment by zvolsky** [comment, marked as answer]
@MilosPaunovic
Yes, your direction is correct.
I have used the header slot. This is not good because it replaces everything include the selection checkbox.
The correct one is v-slot:header-cell or v-slot:header-cell-name (the 2nd is not suitable for me while I have dynamic column names).
The example of v-slot:header-cell is direct in the QTable documentation, it is the one with icon before each column label (icon: opened lock). The example is without selection checkbox. But if I add it (using: selection="multiple"), everything works well.
Thank you for help.

**Reply by zvolsky** [reply]
So finally I have something like this:
```
      <template v-slot:header-cell="props">
        <q-th :props="props" @mouseenter="thMouseenter" @mouseleave="thMouseleave" @click="thClick">
          {{ props.col.label }}
          <q-icon name="swap_vert" @click="orderingIconClicked" />

```
